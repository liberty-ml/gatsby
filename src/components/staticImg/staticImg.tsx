import * as React from 'react'
import { Link, useStaticQuery, graphql } from 'gatsby'
import {PropsWithChildren, ReactElement} from 'react'
import classnames from 'classnames'
import * as styles from './staticImg.module.scss'
import { StaticImage } from "gatsby-plugin-image"

interface StaticImgProps extends PropsWithChildren<any> {

    classNames?: string

    id: string

    alt: string

    src: string
}

export function StaticImg({
  classNames,
  id,
  alt,
  src,
  children
}: StaticImgProps): ReactElement {

  return <StaticImage src="../../images/headshot.jpg" alt="A dinosaur" />
}

StaticImg.displayName = 'StaticImg'
