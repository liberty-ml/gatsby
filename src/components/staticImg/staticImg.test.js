import React from "react"
import renderer from "react-test-renderer"
import { StaticImg } from './staticImg'

describe("StaticImg", () => {
  it("renders correctly", () => {


  })
})
