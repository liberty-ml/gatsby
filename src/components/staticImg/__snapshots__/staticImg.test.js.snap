// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticImg renders correctly 1`] = `
<div
  className=" gatsby-image-wrapper"
  style={
    Object {
      "maxHeight": null,
      "maxWidth": null,
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={true}
    style={
      Object {
        "paddingBottom": "74.66666666666667%",
        "width": "100%",
      }
    }
  />
  <img
    alt=""
    aria-hidden={true}
    src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAPABQDASIAAhEBAxEB/8QAFwAAAwEAAAAAAAAAAAAAAAAAAAQFA//EABQBAQAAAAAAAAAAAAAAAAAAAAH/2gAMAwEAAhADEAAAAZbk9wUzMT//xAAdEAACAQQDAAAAAAAAAAAAAAABAwIABBAREhMh/9oACAEBAAEFAoely+vEZcS+4DDuv//EABURAQEAAAAAAAAAAAAAAAAAABBB/9oACAEDAQE/AYf/xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AT//xAAYEAACAwAAAAAAAAAAAAAAAAARIQABEP/aAAgBAQAGPwISmThipZ//xAAZEAEAAwEBAAAAAAAAAAAAAAABABFBECH/2gAIAQEAAT8hsILWOfCjIscRjCQ1DeP/2gAMAwEAAgADAAAAEIDP/8QAFhEBAQEAAAAAAAAAAAAAAAAAEQEQ/9oACAEDAQE/EILP/8QAFhEBAQEAAAAAAAAAAAAAAAAAEQEQ/9oACAECAQE/EKuf/8QAGBABAAMBAAAAAAAAAAAAAAAAAQARIUH/2gAIAQEAAT8Q1AFAdZZ2gsJT0i3BetBmMBbw32Op/9k="
    style={
      Object {
        "height": "100%",
        "left": 0,
        "objectFit": "cover",
        "objectPosition": "center",
        "opacity": 1,
        "position": "absolute",
        "top": 0,
        "transitionDelay": "500ms",
        "width": "100%",
      }
    }
  />
  <picture>
    <source
      sizes="(max-width: 300px) 100vw, 300px"
      srcSet="/static/71548e2ebe9a7c8691a5b3c9d07bee7d/25b50/headshot.jpg 75w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/aabdf/headshot.jpg 150w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/9dc27/headshot.jpg 300w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/32fd5/headshot.jpg 450w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/4fe8c/headshot.jpg 600w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/ec6c5/headshot.jpg 1280w"
    />
    <img
      alt=""
      loading="lazy"
      onLoad={[Function]}
      sizes="(max-width: 300px) 100vw, 300px"
      src="/static/71548e2ebe9a7c8691a5b3c9d07bee7d/9dc27/headshot.jpg"
      srcSet="/static/71548e2ebe9a7c8691a5b3c9d07bee7d/25b50/headshot.jpg 75w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/aabdf/headshot.jpg 150w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/9dc27/headshot.jpg 300w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/32fd5/headshot.jpg 450w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/4fe8c/headshot.jpg 600w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/ec6c5/headshot.jpg 1280w"
      style={
        Object {
          "height": "100%",
          "left": 0,
          "objectFit": "cover",
          "objectPosition": "center",
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transition": "opacity 500ms",
          "width": "100%",
        }
      }
    />
  </picture>
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "<picture><source srcset=\\"/static/71548e2ebe9a7c8691a5b3c9d07bee7d/25b50/headshot.jpg 75w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/aabdf/headshot.jpg 150w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/9dc27/headshot.jpg 300w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/32fd5/headshot.jpg 450w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/4fe8c/headshot.jpg 600w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/ec6c5/headshot.jpg 1280w\\" sizes=\\"(max-width: 300px) 100vw, 300px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 300px) 100vw, 300px\\" srcset=\\"/static/71548e2ebe9a7c8691a5b3c9d07bee7d/25b50/headshot.jpg 75w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/aabdf/headshot.jpg 150w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/9dc27/headshot.jpg 300w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/32fd5/headshot.jpg 450w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/4fe8c/headshot.jpg 600w,
/static/71548e2ebe9a7c8691a5b3c9d07bee7d/ec6c5/headshot.jpg 1280w\\" src=\\"/static/71548e2ebe9a7c8691a5b3c9d07bee7d/9dc27/headshot.jpg\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
      }
    }
  />
</div>
`;
